google-ads/src/main/java/com/google/ads/googleads/lib/GoogleAdsClient.java

```diff
@@ -149,16 +168,97 @@ public static Builder newBuilder() {
      */
     public abstract Builder setCredentials(Credentials credentials);
 
-    private void setCredentials(Properties properties) {
-      UserCredentials credentials =
-          UserCredentials.newBuilder()
-              .setClientId(properties.getProperty(ConfigPropertyKey.CLIENT_ID.getPropertyKey()))
-              .setClientSecret(
-                  properties.getProperty(ConfigPropertyKey.CLIENT_SECRET.getPropertyKey()))
-              .setRefreshToken(
-                  properties.getProperty(ConfigPropertyKey.REFRESH_TOKEN.getPropertyKey()))
-              .build();
-      setCredentials(credentials);
+    private void setCredentials(Properties properties) throws IOException {
+      String serviceAccountSecretsPath =
+          ConfigPropertyKey.SERVICE_ACCOUNT_SECRETS_PATH.getPropertyValue(properties);
+      String refreshToken = ConfigPropertyKey.REFRESH_TOKEN.getPropertyValue(properties);
+      // Validates that entries are present for exactly one of installed app/web flow credentials or
+      // service account credentials.
+      boolean hasInstalledAppKeys =
+          INSTALLED_APP_OAUTH_KEYS.stream().anyMatch(k -> k.getPropertyValue(properties) != null);
+      boolean hasServiceAccountKeys =
```

Comment:
Nit: The truth condition is hasInstalledAppKeys ^ hasServiceAccountKeys, so it might be simpler to just say 

if (hasInstalledAppKeys ^ hasServiceAccountKeys ) {
setCredentials(
          hasInstalledAppKeys
              ? createUserCredentials(properties)
              : createServiceAccountCredentials(properties));
} else {
        throw new IllegalArgumentException(
            String.format(
                "Properties for credentials are incorrectly specified. Please modify properties to include entries for only one set of keys -- %s if using installed application/web flow credentials, or %s if using"
                    + " service account credentials.",
                INSTALLED_APP_OAUTH_KEYS, SERVICE_ACCOUNT_OAUTH_KEYS));
}

Reply:
I went back and forth on this and considered trying to combine into one error message, but decided that having distinct errors for "credential keys missing altogether" and "multiple types of credential keys found" outweighed the benefit of less code here.

@nwbirnie , WDYT?

Reply:
It's a good spot, and a good call on the trade-off. I think that having a clearer error case here is valuable enough to cover the cost of the additional code.

Reply:
Thanks, Nick. Will leave this as-is then.